app/javascript/src/components/Dashboard/Notes/constants.js

```diff
@@ -1,50 +1,140 @@
 import * as yup from "yup";
 
+import i18n from "../../../common/i18n";
+
+const t = i18n.t.bind(i18n);
+
 export const NOTES_FORM_INITIAL_FORM_VALUES = {
   title: "",
   description: "",
+  assignedContact: "",
+  tags: [],
 };
 
 export const NOTES_FORM_VALIDATION_SCHEMA = yup.object().shape({
-  title: yup.string().required("Title is required"),
-  description: yup.string().required("Description is required"),
+  title: yup.string().required(t("required.entity", { entity: "Title" })),
+  description: yup
+    .string()
+    .required(t("required.entity", { entity: "Description" })),
+  assignedContact: yup
+    .object()
+    .test(
+      "isValidLabel",
+      t("required.entity", { entity: "Assigned Contact" }),
+      value =>
+        value &&
+        value.label &&
+        typeof value.label === "string" &&
+        value.label.trim() !== ""
+    ),
```

Comment:
Please use the `shape` method here. You will find more information in the [official doc](https://github.com/jquense/yup#objectshapefields-object-nosortedges-arraystring-string-schema). Also make the object `nullable`